### PR TITLE
Regression tests for 73 (distutil import error)

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refer to https://github.com/PyCQA/astroid/pull/1321

### CODE REVIEW

File: tests/functional/d/distutil_regression_import_error.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# pylint: disable=missing-docstring
```
Comment (DanielNoord) [inline tests/functional/d/distutil_regression_import_error.py:1]: We might as well add a module docstring + link to some of the relevant issue 😄 

Saves us the trouble of looking them up again when we need to revisit

Comment (cdce8p) [reply tests/functional/d/distutil_regression_import_error.py:1]: I agree. These once could be useful
* https://github.com/PyCQA/pylint/issues/73
* https://github.com/PyCQA/pylint/issues/2955
* https://github.com/PyCQA/astroid/pull/1321

Comment (Pierre-Sassoulas) [reply tests/functional/d/distutil_regression_import_error.py:1]:
```suggestion
"""
Regression test to check that distutils can be imported
See https://github.com/PyCQA/pylint/issues/73

See also:
https://github.com/PyCQA/pylint/issues/2955
https://github.com/PyCQA/astroid/pull/1321
"""
```

File: tests/functional/d/distutil_regression_import_error.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+import distutils.version
+
+print(distutils.version)
```
Comment (Pierre-Sassoulas) [inline tests/functional/d/distutil_regression_import_error.py:4]:
```suggestion
import distutils.version
from distutils.util import strtobool

print(distutils.version)
print(strtobool)
```

**Review by cdce8p — APPROVED** [review]

**Review by cdce8p — COMMENTED** [review]
Was a bit quick there with the approval. Would it make sense to rename the test file? Most other regression tests start with `regression_` I believe.

File: tests/functional/d/distutil_regression_import_error.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+See also:
+https://github.com/PyCQA/pylint/issues/2955
+https://github.com/PyCQA/astroid/pull/1321
+"""
+import distutils.version
+from distutils.util import strtobool
+
+print(distutils.version)
+print(strtobool)
```
Comment (DanielNoord) [inline tests/functional/d/distutil_regression_import_error.py:13]: DIdn't we also want to have a test to see that `distutils` still emitted `no-member` for incorrect imports? We might want to add one case here.

File: tests/functional/d/distutil_regression_import_error.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+"""
+Regression test to check that distutils can be imported
```
Comment (DanielNoord) [inline tests/functional/d/distutil_regression_import_error.py:2]:
```suggestion
"""Regression test to check that distutils can be imported
```

We need to find some pre-commit hook that starts to enforce this.
Perhaps we can look at something for `Path` at the same time? 😄

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: tests/functional/d/distutil_regression_import_error.py (hunk 4 above)
Comment (Pierre-Sassoulas) [reply tests/functional/d/distutil_regression_import_error.py:2]: Maybe https://github.com/PyCQA/pydocstyle ?

Comment (Pierre-Sassoulas) [reply tests/functional/d/distutil_regression_import_error.py:2]: Better yet https://github.com/myint/docformatter

Comment (cdce8p) [reply tests/functional/d/distutil_regression_import_error.py:2]: For module docstrings I think both are fine. Those are tests after all.

> Perhaps we can look at something for Path at the same time? 😄

I wouldn't recommend that. It's not really worth it and might limit us more than it actually would help.

Comment (DanielNoord) [reply tests/functional/d/distutil_regression_import_error.py:2]: > I wouldn't recommend that. It's not really worth it and might limit us more than it actually would help.

This was a little joke as @Pierre-Sassoulas keeps having to remind me to use it on the PRs I open 😄 
Don't think we should enforce it all the time indeed.

Perhaps those packages allow us to exclude modules? Oh well, something to explore another time.

Comment (Pierre-Sassoulas) [reply tests/functional/d/distutil_regression_import_error.py:2]: Yes we definitely don't want to refactor all the old code that uses os.path.

File: tests/functional/d/distutil_regression_import_error.py (hunk 3 above)
Comment (Pierre-Sassoulas) [reply tests/functional/d/distutil_regression_import_error.py]: 22524d2

File: pylint/typing.py [hunk 5]
```diff
@@ -57,7 +57,7 @@ class MessageLocationTuple(NamedTuple):
 
 
 class ManagedMessage(NamedTuple):
-    """Tuple with information ahout a managed message of the linter"""
+    """Tuple with information about a managed message of the linter"""
```
Comment (DanielNoord) [inline pylint/typing.py:60]: Perhaps clean up the commit history and don't squash?

**Comment by Pierre-Sassoulas** [comment]
> Would it make sense to rename the test file?

I renamed :) Although I think having a nice explicit name is better than "regression_1234" so I kept the explicit part.

**Review by DanielNoord — APPROVED** [review]

File: pylint/typing.py (hunk 5 above)
Comment (Pierre-Sassoulas) [reply pylint/typing.py:60]: You're right I wanted to rebase originally but there are other commits that prevents that now.

**Pull request merged**